### PR TITLE
trend: choose trend crash

### LINKED ISSUE #95447: Some visualization panels are unavailable in version 11.3
### What happened?

Start method docker Start command: `docker run -d --name grafana -p 30000:3000 grafana/grafana:11.3.0`

1. Trend
Image
2. Flame Graph
Image
3. XY Chart
Image

The F12 error is as follows
Image

Measures already taken
1. Install Trend separately, command grafana-cli plugins install snuids-trend-panel
2. Delete the container and restart
Thank you

### What did you expect to happen?

Can be used normally

### Did this work before?

First time use

### How do we reproduce it?

Image

### Is the bug inside a dashboard panel?

Yes

### Environment (with versions)?

Grafana: 11.3
OS: docker
Browser:  chrome

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by yincongcyincong**
Fixes #95447

### CODE REVIEW

**Comment by yincongcyincong** [comment]
@Develer @baldm0mma @adela-almasan could you please take a look this pr. thanks.

File: public/app/plugins/panel/trend/TrendPanel.tsx [hunk 1]
```diff
@@ -61,7 +60,7 @@ export const TrendPanel = ({
     } else {
       // first number field
       // Perhaps we can/should support any ordinal rather than an error here
-      xFieldIdx = frames[0].fields.findIndex((f) => f.type === FieldType.number);
+      xFieldIdx = frames[0] ? frames[0].fields.findIndex((f) => f.type === FieldType.number) : -1;
```
Comment (Develer) [inline public/app/plugins/panel/trend/TrendPanel.tsx:63]: I don't see any harm in adding this condition, but I'm worried that the problem can be in some different place. I mean why have empty frames in the first place?

**Comment by yincongcyincong** [comment]
@adrapereira could you please take a look this issue. i think this is same as [#95604](https://github.com/grafana/grafana/pull/95604)

File: public/app/plugins/panel/trend/TrendPanel.tsx (hunk 1 above)
Comment (yincongcyincong) [reply public/app/plugins/panel/trend/TrendPanel.tsx:63]: @Develer bro. do you think this pr can merge or not? or you have some suggestion. i will be pleasure to modified it.

**Review by Develer — APPROVED** [review]
LGTM 🚀

**Comment by yincongcyincong** [comment]
> LGTM 🚀

@Develer Could you please merge this pr? thanks!

**Comment by gelicia** [comment]
@yincongcyincong I just approved the build to run, I'll monitor and merge when ready. Thank you for your patience!

**Pull request merged**